cmd/acra-server/acra-server.go

```diff
@@ -313,7 +320,21 @@ func main() {
 
 		ocspDbVerifier := network.DefaultOCSPVerifier{Config: *ocspDbConfig, Client: &network.DefaultOCSPClient{}}
 
-		dbTLSConfig, err = network.NewTLSConfig(network.SNIOrHostname(*tlsDbSNI, *dbHost), *tlsDbCA, *tlsDbKey, *tlsDbCert, tls.ClientAuthType(*tlsDbAuthType), ocspDbVerifier, crlVerifier)
+		var crlDbConfig *network.CRLConfig
+		if len(*tlsCrlDbUrl) > 0 {
+			crlDbConfig, err = network.NewCRLConfig(*tlsCrlDbUrl, *tlsCrlFromCert)
+		} else {
+			crlDbConfig, err = network.NewCRLConfig(*tlsCrlUrl, *tlsCrlFromCert)
+		}
+		if err != nil {
+			log.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorWrongConfiguration).
+				Errorf("Configuration error: invalid CRL config")
+			os.Exit(1)
+		}
+
+		crlDbVerifier := network.DefaultCRLVerifier{Config: *crlDbConfig, Client: network.DefaultCRLClient{}, Cache: &network.DefaultCRLCache{}}
+
+		dbTLSConfig, err = network.NewTLSConfig(network.SNIOrHostname(*tlsDbSNI, *dbHost), *tlsDbCA, *tlsDbKey, *tlsDbCert, tls.ClientAuthType(*tlsDbAuthType), ocspDbVerifier, crlDbVerifier)
```

Comment:
soon we will need some config for TlsConfig creation )) because too much parameters)

Reply:
> soon we will need some config for TlsConfig creation

Well, I'd say it will be much more readable to have some `network.TLSConfig` with exported, named fields that you fill out, and then use
```go
func (c *TLSConfig) ToTLSConfig() (*tls.Config, error)
```
to convert that into Go structure while making all the necessary validation.

Reply:
Using fields directly less flexible than to use methods, IMHO. For example configuring VerifyConnection handler. How we should configure this field with 2 handlers? Who is responsible to do this as chain of handlers? First added handler, second?
```
config.VerifyConnection = handler1
// some code
if config.VerifyConnection != nil {
   config.VerifyConnection = HandlerChain(config.VerifyConnection, handler2)
} else {
  config.VerifyConnection = handler2
}
```
yeah, it is an exaggerated example, but the main reason that better to encapsulate some logic with methods or in other way to allow change it in a future without rewriting client's code (in our case it's main.go functions in acra-ce and in acra-ee after) if we will change this logic

Reply:
Well... You can go
```go
config := &network.TLSConfig{
        // ...
        Verifier: VerifyAll(verifier1, verifier2),
}
```
I.e., build the verifier before even getting to constructing a TLS config. I doubt that it will be necessary to change the configuration midway, it seems mostly static thing. The struct just gives proper names to components instead of them being a long string of unnamed constructor parameters.

Of course, the builder pattern can be used, but I don't feel that it's that popular in Go, and in this particular case of setting up TLS verification, it seems more natural to keep verifier combinators separate from TLS config rather than having something like
```go
config.AddVerifier(verifier1)
config.AddVerifier(verifier2)
```
because you can never predict all the ways you'd want to combine the verifiers and implement the assortment of methods for setting them up in TLS config itself. It's more flexible to have the combinators freestanding and then just pass the resulting complete verifier to the TLS config.

Reply:
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis - good article with pros/cons of several methods of configuration. Cons for using config as struct is empty values and default configuration.
Actually, I prefer to try options) But anyway we need to use the same wrappers as you show like `VerifyAll(...)`